src/shared/environment-data-sources/ecwid.ts

```diff
@@ -1,41 +1,60 @@
 
 const ecwidTracker = ({ transactionEvent }: Partial<EnvironmentEvents>) => {
   if (!window.transactionOrder && !window.transactionItems) {
     return;
   }
 
-  try {
-    const transaction = tryParseJSONObject(window.transactionOrder);
-    const products = tryParseJSONObject(window.transactionItems);
+  pollForElement(
+    [
+      ".ec-confirmation__number.ec-header-h5",
+      ".ec-confirmation__order-confirmation-total",
+      ".ec-cart-item__title",
+      ".ec-cart-item__count-inner",
+    ],
+    () => {
+      if (window.location.href.includes("/orderConfirmation") || window.location.href.includes("/order-confirmation")) {
+        try {
+          const idElement = document.querySelector(".ec-confirmation__number.ec-header-h5");
+          const totalElement = document.querySelector(".ec-confirmation__order-confirmation-total");
+          const nameElement = document.querySelector(".ec-cart-item__title");
+          const quantityElement = document.querySelector(".ec-cart-item__count-inner");
 
-    transactionEvent({
-      id: transaction.id.toString(),
-      total: parseFloat(transaction.total),
-      tax: parseFloat(transaction.tax) || 0,
-      shipping: parseFloat(transaction.delivery_fee) || 0,
-      city: "N/A",
-      state: "N/A",
-      country: "USA",
-      currency: "USD",
-      items: products.map((product) => {
-        const { item_id, item_name, item_category, price, quantity } = product;
-        return {
-          orderId: transaction.id.toString(),
-          productId: item_id.toString(),
-          sku: item_id.toString(),
-          name: (item_name || "N/A").toString(),
-          category: (item_category || "N/A").toString(),
-          unitPrice: parseFloat(price || 0),
-          quantity: parseInt(quantity || 1),
-          currency: "USD",
-        } as TransactionCartItem;
-      }),
-    });
-  } catch (error) {
-    window.tracker('trackError', JSON.stringify(error), 'ECWID');
-  }
+          var id = idElement.textContent;
+          var total = totalElement.textContent.replace("$", "");
+          var name = nameElement.textContent;
+          var quantity = quantityElement.textContent.replace("× ", "");
+
+          const products: TransactionCartItem[] = [
+            {
+              orderId: id.toString(),
+              sku: "N/A",
+              name: (name || "N/A").toString(),
+              category: "N/A",
+              unitPrice: parseFloat(total) || 0,
+              quantity: parseInt(quantity) || 1,
+              currency: "USD",
+            },
+          ];
+
+          transactionEvent({
+            id: id.toString(),
+            total: parseFloat(total),
+            tax: 0,
+            shipping: 0,
+            city: "N/A",
+            state: "N/A",
+            country: "USA",
+            currency: "USD",
+            items: products,
+          });
+        } catch (error) {
+          window.tracker("trackError", JSON.stringify(error), "ECWID");
```

Comment:
This is great but you'll need to import this to the v1 and v2 carts folder to use it. It's not quite doing anything right now